account_invoice_report_grouped_by_picking/security/portal_security.xml

```diff
@@ -1,16 +1,16 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8" ?>
 <odoo noupdate="1">
-
     <!-- Stock Portal Access Rules for commercial partner -->
     <record id="portal_stock_picking_portal_commercial_partner_rule" model="ir.rule">
         <field name="name">Portal Commercial Pickings</field>
-        <field name="model_id" ref="stock.model_stock_picking"/>
-        <field name="domain_force">['|', ('partner_id.id', 'child_of', user.partner_id.commercial_partner_id.id), ('sale_id.partner_id.id', 'child_of', user.partner_id.commercial_partner_id.id)]</field>
-        <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
-        <field name="perm_read" eval="True"/>
-        <field name="perm_write" eval="False"/>
-        <field name="perm_create" eval="False"/>
-        <field name="perm_unlink" eval="False"/>
+        <field name="model_id" ref="stock.model_stock_picking" />
+        <field
+            name="domain_force"
+        >['|', ('partner_id.id', 'child_of', user.partner_id.commercial_partner_id.id), ('sale_id.partner_id.id', 'child_of', user.partner_id.commercial_partner_id.id)]</field>
```

Comment:
I don't think this is the best approach for avoiding the access problem, as what happened to a public user with a token for accessing the invoice?

I think the solution is to put sudo where needed in the QWeb report. Note that this problem is also in v12, but I have detected it now.

Reply:
OH! yes!
https://github.com/odoo/odoo/blob/60704b3151f66b4daf87c147afcd231ae0713ab3/addons/account/controllers/portal.py#L82
The invoices are rendered as sudo, so we can remove this rule

Reply:
In v12 we can remove it too